File: low-level-design/design-patterns/python/creational/abstract-factory/gcp_factory.py
from factory import Factory

from gcp import GCPCompute,GCPNetwork,GCPStorage

class GCPFactory(Factory):
    def createCompute(self):
        return GCPCompute()
    def createStorage(self):
        return GCPStorage()
    def createNetwork(self):
        return GCPNetwork()

File: low-level-design/design-patterns/golang/creational/abstract-factory/abstract-factory/gcp_factory.go
package abstractfactory

func NewGCPFactory() *GCPFactory {

	return &GCPFactory{}
}

type GCPFactory struct {
}

func (f *GCPFactory) CreateCompute() Compute {
	return NewGCPCompute()
}

func (f *GCPFactory) CreateStorage() Storage {
	return NewGCPStorage()
}

func (f *GCPFactory) CreateNetwork() Network {

	return NewGCPNetwork()
}
